.github/workflows/ci.yml

```diff
@@ -0,0 +1,94 @@
+name: CI
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+env:
+  IMAGE_TAG: ${{ github.sha }}
+jobs:
+  test:
+    runs-on: ubuntu-latest
```

Comment:
why not alpine?

Reply:
There's a limited set of github pipeline environments to choose from and they don't mirror docker container images https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idruns-on

Reply:
Also because they are not containers: they are Azure virtual machines.